### PR TITLE
feat: `cloudavenue_vm`

### PR DESCRIPTION
**Pull request by azrod**
### Description of your changes
Draft PR to add `cloudavenue_vm`resource and fix `cloudavenue_vm_disk` resource

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### CODE REVIEW

File: examples/resources/cloudavenue_vm_disk/resource.tf [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-    bus_number  = 0
-    unit_number = 1
-  }
+resource "cloudavenue_vapp" "example" {
+  vapp_name   = "vapp_example"
+  description = "This is a example vapp"
+}
+
+resource "cloudavenue_vm_disk" "example-detachable" {
+  vapp_id       = "your_value"
```
[the remark this answers is not in the document]
Comment (azrod) [reply examples/resources/cloudavenue_vm_disk/resource.tf:7]: This is automaticaly replaced by tf-doc-extractor

File: internal/provider/common/vm/diskparams/bus_type.go [hunk 2]
```diff
@@ -1,34 +1,118 @@
...
+		MarkdownDescription: budTypeDescription,
+	}
+}
+
+// BusTypeAttributeRequired returns a schema.Attribute with a required value.
+func BusTypeAttributeRequired() schema.Attribute {
 	return schema.StringAttribute{
 		Required:            true,
-		MarkdownDescription: "The type of disk controller. Possible values: `ide`, `parallel` (LSI Logic Parallel SCSI), `sas` (LSI Logic SAS SCSI), `paravirtual` (Paravirtual SCSI), `sata`, `nvme`.",
+		MarkdownDescription: budTypeDescription,
```
Comment (eremid) [inline internal/provider/common/vm/diskparams/bus_type.go:113]: 🍺

**Review by eremid — CHANGES_REQUESTED** [review]

File: internal/provider/common/vm/resource.go [hunk 3]
```diff
@@ -19,38 +28,132 @@ type Resource struct {
...
+	var resource Resource
+
+	if vm.VM.VmSpecSection != nil {
+		// CPU
+		resource.CPUs = types.Int64Value(int64(*vm.VM.VmSpecSection.NumCpus))
+		resource.CPUCores = types.Int64Value(int64(*vm.VM.VmSpecSection.NumCoresPerSocket))
+
+		// Memory
+		if vm.VM.VmSpecSection.MemoryResourceMb != nil {
+			resource.Memory = types.Int64Value((*&vm.VM.VmSpecSection.MemoryResourceMb.Configured))
```
Comment (eremid) [inline internal/provider/common/vm/resource.go:148]: *&vm will be simplified to vm.

File: internal/provider/common/vm/resource.go [hunk 4]
```diff
@@ -19,38 +28,132 @@ type Resource struct {
...
+		}
+	}
+
+	if vm.VM.VMCapabilities != nil {
+		// HotAddEnabled
+		resource.CPUHotAddEnabled = types.BoolValue(*&vm.VM.VMCapabilities.CPUHotAddEnabled)
+		resource.MemoryHotAddEnabled = types.BoolValue(*&vm.VM.VMCapabilities.MemoryHotAddEnabled)
```
Comment (eremid) [inline internal/provider/common/vm/resource.go:155]: *&vm will be simplified to vm.

File: internal/provider/common/vm/customization.go [hunk 5]
```diff
@@ -124,3 +198,30 @@ func CustomizationSchema() map[string]schema.Attribute {
 		},
 	}
 }
+
+// CustomizationRead reads the customization block from the API response.
+func CustomizationRead(vm *govcd.VM) (c Customization, err error) {
+	customizationSection, err := vm.GetGuestCustomizationSection()
+	if err != nil {
+		return c, fmt.Errorf("unable to get guest customization section: %s", err)
```
Comment (eremid) [inline internal/provider/common/vm/customization.go:206]: return c, fmt.Errorf("unable to get guest customization section: %w", err)

File: internal/provider/vm/common_vm.go [hunk 6]
```diff
@@ -91,90 +117,99 @@ func updateGuestCustomizationSetting(v *VMClient, vm *govcd.VM) error {
...
 
-	cust := v.Plan.Customization[0]
+	var customization commonvm.Customization
+	v.Plan.Customization.As(context.Background(), customization, basetypes.ObjectAsOptions{})
 
-	if !cust.Force.IsNull() && !cust.Force.IsUnknown() {
-		return cust.Force.ValueBool()
+	if !customization.Force.IsNull() && !customization.Force.IsUnknown() {
+		return customization.Force.ValueBool()
 	} else {
```
Comment (eremid) [inline internal/provider/vm/common_vm.go:173]:
```suggestion
}
return false
```

File: internal/provider/vm/common_vm_schema_disk.go [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	if vm.VM != nil && vm.VM.VmSpecSection != nil && vm.VM.VmSpecSection.DiskSection != nil && vm.VM.VmSpecSection.DiskSection.DiskSettings != nil {
+		disks := vmResourceModelDisks{}
+
+		for _, disk := range vm.VM.VmSpecSection.DiskSection.DiskSettings {
+			vmResourceModelDisk := DiskFromGovcdDiskSettings(disk)
+			disks.Append(vmResourceModelDisk)
+		}
+
+		return disks, nil
+	} else {
```
Comment (eremid) [inline internal/provider/vm/common_vm_schema_disk.go:173]:
```suggestion
	}
```

File: internal/provider/vm/common_vm_schema_disk.go [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+		"name":            types.StringType,
+		"bus_type":        types.StringType,
+		"size_in_mb":      types.Int64Type,
+		"storage_profile": types.StringType,
+		"bus_number":      types.Int64Type,
+		"unit_number":     types.Int64Type,
+	}
+}
+
+func (x *vmResourceModelDisks) ObjectType() attr.Type {
```
Comment (eremid) [inline internal/provider/vm/common_vm_schema_disk.go:45]:
```suggestion
func (d *vmResourceModelDisks) ObjectType() attr.Type {
```

**Review by eremid — CHANGES_REQUESTED** [review]

**Pull request merged**